tests/tests/eq_mpz.plf

```diff
@@ -0,0 +1,7 @@
+(declare bool type)
+(declare tt bool)
+(declare ff bool)
+
+
+(program eq_mpz ((x mpz) (y mpz) (b bool)) bool
+  (ifequal x y tt ff))
```

Comment:
Maybe expand the test to actually invoke the side condition to make sure that it doesn't just type check but actually works as intended? Also `(b bool)` above seems to be unused.